src/effects/lv2/lv2effectprocessor.cpp

```diff
@@ -13,11 +13,11 @@ LV2EffectProcessor::LV2EffectProcessor(EngineEffect* pEngineEffect,
           m_audioPortIndices(audioPortIndices),
           m_controlPortIndices(controlPortIndices),
           m_pEffectsManager(nullptr) {
-    m_inputL = new float[MAX_BUFFER_LEN];
-    m_inputR = new float[MAX_BUFFER_LEN];
-    m_outputL = new float[MAX_BUFFER_LEN];
-    m_outputR = new float[MAX_BUFFER_LEN];
-    m_params = new float[pManifest->parameters().size()];
+    m_inputL = std::vector<CSAMPLE>(MAX_BUFFER_LEN);
+    m_inputR = std::vector<CSAMPLE>(MAX_BUFFER_LEN);
+    m_outputL = std::vector<CSAMPLE>(MAX_BUFFER_LEN);
+    m_outputR = std::vector<CSAMPLE>(MAX_BUFFER_LEN);
+    m_params = std::vector<float>(pManifest->parameters().size());
```

Comment:
I don't really know actually. If we rely on the fact that we don't reallocate later, we should use std::array. If we rather want to save some memory, std::vector might be the better choice.

Reply:
If the size is known at compile time then we could use std::array.

Reply:
Well, in this particular case, its known because its a `#define`. However, I don't know if thats because we actually know that there is only ever going to be exactly `MAX_BUFFER_LEN` elements, or if we just allocate this many elements statically to make sure we don't have to reallocate in RT code.

Reply:
MAX_BUFFER_LEN is an ugly quick hack. The buffers should be allocated based on the buffer size configured in the preferences and reallocated when that is changed. That is outside of the scope of this PR, but for this PR we should not rely on the buffer size being known at compile time.

Reply:
Well its standard practice in microcontroller-realm where there is no heap-allocation. I assume this code was written by daniel which is why he has less issues with this method than others.